api/v1alpha1/nodehealthcheck_types.go

```diff
@@ -101,70 +119,163 @@ type NodeHealthCheckSpec struct {
+	// RemediationTemplate is a reference to a remediation template
+	// provided by a remediation provider.
+	//
+	// If a node needs remediation the controller will create an object from this template
+	// and then it should be picked up by a remediation provider.
+	//
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	RemediationTemplate corev1.ObjectReference `json:"remediationTemplate"`
+
+	// Order defines the order for this remediation.
+	// Remediations with lower order will be used before remediations with higher order.
+	// Remediations must not have the same order.
+	//
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	Order int `json:"order"`
+
+	// Timeout defines how long NHC will wait for the node getting healthy
+	// before the next remediation (if any) will be used. When the last remediation times out,
+	// the overall remediation is considered as failed.
+	// As a safeguard for preventing parallel remediations, a minimum of 60s is enforced.
+	//
+	// Expects a string of decimal numbers each with optional
+	// fraction and a unit suffix, eg "300ms", "1.5h" or "2h45m".
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	//
+	//+kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
+	//+kubebuilder:validation:Type=string
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	Timeout metav1.Duration `json:"timeout"`
+}
+
 // NodeHealthCheckStatus defines the observed state of NodeHealthCheck
 type NodeHealthCheckStatus struct {
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="observedNodes",xDescriptors="urn:alm:descriptor:com.tectonic.ui:observedNodes"
-	//ObservedNodes specified the number of nodes observed by using the NHC spec.selector
-	// +kubebuilder:default:=0
+	// ObservedNodes specified the number of nodes observed by using the NHC spec.selector
+	//
+	//+kubebuilder:default:=0
+	//+operator-sdk:csv:customresourcedefinitions:type=status
 	ObservedNodes int `json:"observedNodes"`
 
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="healthynodes",xDescriptors="urn:alm:descriptor:com.tectonic.ui:healthyNodes"
-	//HealthyNodes specified the number of healthy nodes observed
-	// +kubebuilder:default:=0
+	// HealthyNodes specified the number of healthy nodes observed
+	//
+	//+kubebuilder:default:=0
+	//+operator-sdk:csv:customresourcedefinitions:type=status
 	HealthyNodes int `json:"healthyNodes"`
 
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="inFlightRemediations",xDescriptors="urn:alm:descriptor:com.tectonic.ui:inFlightRemediations"
-	//InFlightRemediations records the timestamp when remediation triggered per node
+	// UnhealthyNodes tracks currently unhealthy nodes and their remediations.
+	//
+	//+patchStrategy=merge
+	//+patchMergeKey=type
+	//+optional
+	//+operator-sdk:csv:customresourcedefinitions:type=status
+	UnhealthyNodes []*UnhealthyNode `json:"unhealthyNodes,omitempty"`
```

Comment:
What's the failure mode if an admin destroys or otherwise messes this section up?

Reply:
even if we would try to use this status less for making decisions, and e.g. look at remediation CRs more for cheking the cluster state, admins can mess up those CRs 🤷🏼‍♂️